src/main/scala/rps/Game.scala

```diff
@@ -1,47 +1,37 @@
 package rps
 
-import RPSMoves._
+import RPSMove._
 import io.buildo.enumero.{CaseEnumIndex, CaseEnumSerialization}
 
 
 
-object Game {
-  def play(): Unit = {
 
+object Game {
+  def play(userMoveString: String): Option[RPSResponse] = {
     /* Ask for user input */
-    val userMoveString = readLine(s"""
-    |What's your move?
-    |0: Rock
-    |1: Paper
-    |2: Scissors
-    |
-    |> """.stripMargin)
-
-    val userMove = CaseEnumIndex[RPSMoves].caseFromIndex(userMoveString)
+    val userMove = CaseEnumSerialization[RPSMove].caseFromString(userMoveString)
 
     userMove match {
-      case None => println(s"Mmm looks like your move was not legal... 🤔")
+      case None => None
       case Some(userMove) => {
 
         val computerMove = generateCPUMove()
-        println(s"Your move was:    ${CaseEnumSerialization[RPSMoves].caseToString(userMove)}")
-        println(s"The CPU move was: ${CaseEnumSerialization[RPSMoves].caseToString(computerMove)}")
+        val outcome = computeGameOutcome(userMove, computerMove)
 
-        /* Compute the outcome */
-        computeGameOutcome(userMove, computerMove)
+        Some(RPSResponse(CaseEnumSerialization[RPSMove].caseToString(userMove), CaseEnumSerialization[RPSMove].caseToString(computerMove), outcome))
```

Comment:
If you end up doing

```scala
x match {
  case None => None
  case Some(x) => Some(f(x))
}
```

you can use `.map` instead:

```scala
x.map(f)
```

Reply:
IIRC you can do `RPSMove.caseToString(userMove)`, but you don't need any of this. See my comment below.